service/matching/matcher.go

```diff
@@ -61,7 +61,8 @@ const (
 	_defaultTaskDispatchRPSTTL = 60 * time.Second
 )
 
-var errTasklistThrottled = errors.New("cannot add to tasklist, limit exceeded")
+// ErrTasklistThrottled implies a tasklist was throttled from the client-side
```

Comment:
hmm.  throttled how?
is this the "10/s global tasklist response-limit" throttle that client-library-users can set in their worker config, or something else?

Reply:
I was under the impression this was tasklist based throttling set in client config?

Reply:
pretty sure it's not:
```go
	dPtr := _defaultTaskDispatchRPS
	limiter := quotas.NewRateLimiter(&dPtr, _defaultTaskDispatchRPSTTL, config.MinTaskThrottlingBurstSize())
```
^ the only non-constant there is the burst-size argument, and that's coming from dynamic config, not the client AFAICT:
```go
		MinTaskThrottlingBurstSize: func() int {
			return config.MinTaskThrottlingBurstSize(domainName, taskListName, taskType)
		},
```

Reply:
yea, looks like this is just a constant ratelimiter (100,000/s) with configurable burst size (.... that seems a bit nuts with 100k/s, but alright) that can be updated from dynamic config every minute.

feels more like hot-loop protection or something to me.  exceeding 100k/s would probably require both large amounts of data and running fully in memory, and I'm not sure that's feasible here without bugs.

Reply:
my mistake, I do not know precisely. I'll update